### ISSUE TITLE
Changing cmap for Image with label

### ISSUE THREAD

**Issue by vascotenner** [ioam/holoviews#33]
Changeing the cmap of an image of a label does not work well:
No error, cmap changed, but label lost:
```python
Image(data)(style={'cmap':'gray'}, label='A')
```
Error:
```python
Image(data, label='A')(style={'cmap':'gray'})
```
```python
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
<ipython-input-99-094bbee28a14> in <module>()
      8     else:
      9         return far + near + pump
---> 10 show_images((7,8,10,9))

<ipython-input-99-094bbee28a14> in show_images(index)
      4     pump = Image(read_near(files[index[2]]), label='M'+files[index[2]])
      5     if len(index) > 3:
----> 6         dmk = Image(read_dmk(files[index[3]]), label='M'+files[index[3]])(style={'cmap':'gray'})
      7         return far + near + pump + dmk
      8     else:

/usr/local/lib/python2.7/dist-packages/holoviews/core/element.pyc in __call__(self, **kwargs)
    135         for k,d in kwargs.items():
    136             options[identifier][k] = d
--> 137         return super(Element, self).__call__(options)
    138 
    139 

/usr/local/lib/python2.7/dist-packages/holoviews/core/dimension.pyc in __call__(self, options, **kwargs)
    615         """
    616         deep_clone = self.map(lambda x: x.clone(id=x.id))
--> 617         StoreOptions.set_options(deep_clone, options, **kwargs)
    618         return deep_clone
    619 

/usr/local/lib/python2.7/dist-packages/holoviews/core/options.pyc in set_options(cls, obj, options, **kwargs)
   1068         spec, compositor_applied = StoreOptions.expand_compositor_keys(options)
   1069 
-> 1070         custom_trees = StoreOptions.create_custom_trees(obj, spec)
   1071         Store.custom_options.update(custom_trees)
   1072         for tree_id in custom_trees.keys():

/usr/local/lib/python2.7/dist-packages/holoviews/core/options.pyc in create_custom_trees(cls, obj, options)
    961 
    962         return {k:cls.apply_customizations(options, t) if options else t
--> 963                 for k,t in clones.items()}
    964 
    965 

/usr/local/lib/python2.7/dist-packages/holoviews/core/options.pyc in <dictcomp>((k, t))
    961 
    962         return {k:cls.apply_customizations(options, t) if options else t
--> 963                 for k,t in clones.items()}
    964 
    965 

/usr/local/lib/python2.7/dist-packages/holoviews/core/options.pyc in apply_customizations(cls, spec, options)
    899                 customization = {k:(Options(**v) if isinstance(v, dict) else v)
    900                                  for k,v in spec[key].items()}
--> 901             options[str(key)] = customization
    902         return options
    903 

/usr/local/lib/python2.7/dist-packages/holoviews/core/tree.pyc in __setitem__(self, identifier, val)
    147             self.__setattr__(identifier, val)
    148         elif isinstance(identifier, str) and self.parent is None:
--> 149             self.set_path(tuple(identifier.split('.')), val)
    150         elif isinstance(identifier, tuple) and self.parent is None:
    151             self.set_path(identifier, val)

/usr/local/lib/python2.7/dist-packages/holoviews/core/tree.pyc in set_path(self, path, val)
    100 
    101         if not all(p[0].isupper() for p in path):
--> 102             raise Exception("All paths elements must be capitalized.")
    103 
    104         if len(path) > 1:

Exception: All paths elements must be capitalized.
```

**Comment by jlstevens** [comment]
Also very surprising. The second example you mention should also work fine:

```python
import numpy as np
%load_ext holoviews.ipython
from holoviews import Image
Image(np.random.rand(5,5), label='A')(style={'cmap':'gray'})
```

Are you using HoloViews 1.0.1 from PyPI (i.e. installed with pip) or are you using a development version cloned from GitHub?

**Comment by jlstevens** [comment]
Does the object ``far + near + pump + dmk`` display ok (i.e without an error) if you don't try setting the 'gray' color map?

By the way, if you want **all** Images to always to be shown with grayscale (by default) you can do:

```python
Store.options.Curve = Options('style', cmap='gray')
```

Alternatively, you could set a group and set a style specific to that group. E.g if you have:

```python
Image(read_dmk(files[index[3]]), label='M'+files[index[3]], group='DMK')
```

You could set this once for all Images that are to be displayed in the style of ``dmk``:

```python
Store.options.Image.DMK = Options('style', cmap='gray')
```

Hope that helps but I would still like to get to the bottom of your original error!

**Comment by vascotenner** [comment]
The problem is in the label. Labels with '.' seems to be problematic:
```python
Image(data, label='M09_pump.bmp')(style={'cmap':'Greens_r'})
```

Thank you for the suggestion to use groups. This makes the code more readable.

**Comment by jlstevens** [comment]
We will probably just escape the period like everything else (when used for attribute access). This will be fixed by the next bug fix release (i.e very soon).

**Comment by jlstevens** [comment]
This issue should now be fixed in 612f6d3f. I'll keep this issue open for a while longer in case anyone has any comments on this commit.

**Comment by jlstevens** [comment]
Ok. Looks like I can close this issue now.
